demo/demo.md

```diff
@@ -191,74 +178,100 @@ The following example is static to illustrate a error state returned with page l
   <span slot="trigger">See code</span>
 
   ```html
-  <auro-input id="error1" required error>
+  <auro-input id="demo5" required>
     <span slot="label">Name</span>
   </auro-input>
 
-  <auro-input id="error2" required bordered error="Please enter your full name.">
+  <auro-input id="demo6" required bordered>
     <span slot="label">Name</span>
     <span slot="helptext">Please enter your full name.</span>
   </auro-input>
   ```
 </auro-accordion>
 
-When using the `required` property, a custom message can be inserted using the `setCustomValidity` property. The value will override the default message set by the client.
+### Custom pattern validation
```

Comment:
There is an issue with the `pattern` example. When the `error` appears, it replaces the `help text` with the `error`. But when the `error` is cleared, the `help text` does not reappear until the user exits the element, e.g. a `blur`. 

![patternError](https://user-images.githubusercontent.com/181089/191107212-80007acc-bd0a-4e9c-a718-90802234cfbd.gif)

Also, what is the difference between this and the `pattern` demo that is on line `100`?

Reply:
Cleaned up the label. The demo on 192 should show its about using the validateOnInput example not just about validating the pattern.

Reply:
> There is an issue with the `pattern` example. When the `error` appears, it replaces the `help text` with the `error`. But when the `error` is cleared, the `help text` does not reappear until the user exits the element, e.g. a `blur`.

We wouldn't remove the error while focus is in the input unless the `validateOnInput` attribute is used. The Pattern exaple here on line 100 does not use that attribute.

Reply:
**Current experience on the docsite**

![current](https://user-images.githubusercontent.com/181089/191130961-62c32150-ffe9-41bb-a1de-a01ebf9495d9.gif)

**Updated version experience**

See how the `helpText` area is cleared out while I am trying to correct the issue. A user would lose context.  

![updated](https://user-images.githubusercontent.com/181089/191130992-4210540e-81f4-4ce1-b0ba-25ebe584bdee.gif)

Reply:
@blackfalcon, I think I see what you mean now. The issue is that I am always surfacing whatever the html5 input validationMessage is and this updates live (because getErrorMessage() is called in the template) but not running auro-input.validation() until blur. So this puts it in a state where the input is invalid so its trying to show the error message but there is no error message from the html5 input to show since it validated itself live.

I've made an update so that getErrorMessage will persist the last message from input.validationMessage so that it won't change until we have a chance to revalidate.